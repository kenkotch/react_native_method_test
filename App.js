import React from 'react'
import {
  Text,
  TouchableOpacity,
  View
} from 'react-native'

export default class App extends React.Component {
  state = {
    counter: 0
  }

  incrementCounter = number => {
    const counter = this.state.counter + number
    this.setState({ counter })
  }

  resetCounter = () => {
    this.setState({ counter: 0 })
  }

  render() {
    const { counter } = this.state
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Counter: { counter }</Text>
        <TouchableOpacity
          onPress={ () => this.incrementCounter(1) }
        >
          <Text>
            Increment by 1
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={ () => this.incrementCounter(4) }
        >
          <Text>
            Increment by 4
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={ this.resetCounter }
        >
          <Text>
            Reset
          </Text>
        </TouchableOpacity>
      </View>
    )
  }
}

const styles = {
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  }
}
