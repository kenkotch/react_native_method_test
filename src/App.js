import React from 'react'
import { Provider } from 'react-redux'
import store from './store'
import Main from './components/ConnectedMain'

export default class App extends React.Component {
  render() {
    return (
      <Provider store={ store }>
        <Main />
      </Provider>
    )
  }
}
