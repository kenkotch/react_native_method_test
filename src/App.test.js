import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import App from './App'

configure({ adapter: new Adapter() })

describe('App Tests', () => {
  it('App renders the same every time', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
