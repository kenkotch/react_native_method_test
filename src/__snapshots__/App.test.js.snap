// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests App renders the same every time 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    Counter from local state: 
    0
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Increment by 1
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Increment by 4
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Reset
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 15,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    Count from redux state: 
    0
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Increment by 1
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Increment by 5
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Reset
    </Text>
  </View>
</View>
`;
